version: "3.6"

services:
  ulc_db:
    image: postgres:11
    container_name: ulc_db
    environment:
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_DB=${DB_NAME}
    networks:
      - ulc-network
    volumes:
      - ${DB_PATH}:/var/lib/postgresql/data
      - ${DB_DUMP}:/docker-entrypoint-initdb.d/dump.sql

  ulc_redis:
    image: redis
    container_name: ulc_redis
    networks:
      - ulc-network
    expose:
      - "6379"

  ulc_site:
    image: ulc:latest
    container_name: ulc_site
    environment:
      - C_FORCE_ROOT='true'
      - PYTHONUNBUFFERED=1
      - TZ=Europe/Moscow
      - DB_PASS=${DB_PASSWORD}
      - DB_USER=${DB_USER}
      - DB_NAME=${DB_NAME}
    ports:
      - 41999:8000
    depends_on:
      - ulc_db
    networks:
      - ulc-network
    volumes:
      - ${SITE_PATH}:/ulc
    command: ./start.sh

networks:
  ulc-network:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: ${NETWORK_ADDRESS}